### PR TITLE
Keep derivation env field.

### PR DESCRIPTION
**Pull request by jsoo1**

### CODE REVIEW

File: src/Nix/Graph/Internal.hs [hunk 1]
```diff
@@ -52,9 +53,12 @@ data Derivation = Derivation
   , derivationSystem :: Text
   , derivationInputDrvs :: [FilePath]
   , derivationBuilt :: Bool
+  , derivationEnv :: MapS.Map Text Text
   }
   deriving stock (Eq, Ord, Generic)
-  deriving anyclass (Hashable)
+
+instance Hashable Derivation where
+  hashWithSalt s = hashWithSalt s . derivationPath
```
Comment (evanrelf) [inline src/Nix/Graph/Internal.hs]: I think this ignores the `derivationBuilt` field, so we should stick with the derived instance.

File: src/Nix/Graph/Internal.hs [hunk 2]
```diff
@@ -28,6 +28,7 @@ import qualified Control.Concurrent.STM.Map as STM.Map
 import qualified Control.Concurrent.STM.TSem as TSem
 import qualified Data.Attoparsec.Text as Attoparsec
 import qualified Data.Map as Map
+import qualified Data.Map.Strict as MapS
```
Comment (evanrelf) [inline src/Nix/Graph/Internal.hs]: I'd follow how `relude` names things, and call these `LMap` and `Map`:

```suggestion
import qualified Data.Map as LMap
import qualified Data.Map.Strict as Map
```

**Review by evanrelf — APPROVED** [review]
Looks great! 🙂 Don't forget to update the changelog.

File: src/Nix/Graph/Internal.hs (hunk 1 above)
Comment (jsoo1) [reply src/Nix/Graph/Internal.hs]: With the addition of a `Map`, the instance can no longer be derived.

Comment (jsoo1) [reply src/Nix/Graph/Internal.hs]: Will there be duplicates if the derivation is built/unbuilt?

**Pull request merged**